data_collection/gazette/spiders/ba/ba_candeias.py

```diff
@@ -0,0 +1,14 @@
+from datetime import date
+
+from gazette.spiders.base.brtransparencia import BaseBrTransparenciaSpider
+
+
+class BaCandeiasSpider(BaseBrTransparenciaSpider):
+    name = "ba_candeias"
+    TERRITORY_ID = "2906501"
+    allowed_domains = ["www.camaraibicoara.ba.gov.br"]
+    start_urls = ["https://www.camaraibicoara.ba.gov.br/diario.html"]
+    start_date = date(2022, 12, 29)
+    br_tranparencia_entity = "63147391-dcb2-4d6c-9c5a-c4483a9d8306"
```

Comment:
@talesmota normalmente nessas classes de RPA evitamos adicionar parâmetros que não sejam parâmetros-base quando podem ser evitados, penso que deveriamos ser capazes de fazer algum tipo de contulta ou mapeamento prévo em outro lugar para obter esses parâmetros de `br_tranparencia_entity` e `br_tranparencia_code`

Reply:
Esses dois parâmetros, br_tranparencia_entity e br_tranparencia_code, são bem específicos para esse tipo de layout, já que são definidos dentro do JavaScript da página. Por isso, optei por deixá-los para serem definidos nas classes filhas.

Reply:
Pelo que pesquisei, não há uma forma diferente de fazer a busca deste ID ao acessar o site

Reply:
Se eu puder dar um pitaco aqui, se for só uma requisição do arquivo javascript antes de começar a raspagem pra valer acho que vale fazer uns regex e obter os dois parâmetros (se funcionar bem pra todos os municípios do raspador).

Se não, vale documentar como obter os parâmetros em forma de docstring no raspador base.

Reply:
Testando aqui pra Ibicoara e pra Itaquara, realmente seria só uma requisição a mais no início. E dá pra ver que tem diferenças, mas são poucas, então pode ser bem generalizável.

Ambos os arquivos requisitados são `https://<domínio>/js/content.js` mas o regex levaria em conta que podem ou não existir espaços em branco em `entity = ` e `code = `.

Em Ibicoara:
```
//variaveis
var entity="691bea32-9b9f-40f8-ab18-31e079080a1a";
var code="CODE_ENT_CM204";
var url="https://api.brtransparencia.com.br/api";
var url2="https://api.brtransparencia.com.br";

...
```

Em Itaquara:
```
...

//variaveis
var entity = "1557447a-9381-44ad-9c0f-016868769479";
var code = "CODE_ENT_PM003";
var url = "https://api.brtransparencia.com.br/api";
var url2 = "https://api.brtransparencia.com.br";

...
```

Realmente acho que vale a pena conferir se generaliza legal pros outros municípios.

Reply:
Boa @ogecece . Agradeço o feedback. Foi muito bem colocado.

Eu analizei aqui o cenário e fiz alguns testes aqui com essa regex, **var {field}(\ )*=(\ )*["|\'](.+?)["|\'];**

E em ambos os casos ela foi capaz de extrair os códigos:
- var entity = "1557447a-9381-44ad-9c0f-016868769479";
- var entity="1557447a-9381-44ad-9c0f-016868769479";
- var entity   = '1557447a-9381-44ad-9c0f-016868769479';
- var entity   =   '1557447a-9381-44ad-9c0f-016868769479';

Acho que chegamos em  um nível de generalização bacana.

Adicionando o log dos novos testes:

[ba_candeias_2023-01-01_2023-03-02.csv](https://github.com/user-attachments/files/17270753/ba_candeias_2023-01-01_2023-03-02.csv)
[ba_candeias_2023-01-01_2023-03-02.txt](https://github.com/user-attachments/files/17270754/ba_candeias_2023-01-01_2023-03-02.txt)
[5bac9ba5c8aec2e83d74684ca53ac373d85d715b.pdf](https://github.com/user-attachments/files/17270755/5bac9ba5c8aec2e83d74684ca53ac373d85d715b.pdf)